src/malli/core.cljc

```diff
@@ -232,6 +232,14 @@
   (let [{min'' :min max'' :max} (-regex-min-max child)]
     (cond-> {:min (f (or min' 0) min'')} (and max' max'') (assoc :max (f max' max'')))))
 
+(defn- -re-cat-min-max
```

Comment:
anything that is not a regex schema takes on slot. e.g. `[:cat [:and [:cat :int :int]]]`.

Reply:
That's handled by the default `RegexSchema` instance `-regex-min-max` here: 

https://github.com/metosin/malli/pull/851/files#diff-c10cf1bd7f29baf493dfd1fb727c955f2bf09a256cdc7ff690343b5940f14940L125

There are also existing tests for this behaviour, which my change doesn't break.

Reply:
failing case:

```clojure
(m/-regex-min-max
 (m/schema
  [:cat {:registry {::cat [:cat string? int?]}}
   int? ::cat]))
; => {:min 3, :max 3}
```

There are currently one `RefSchema` that is also `RegexSchema`, which is the one created with `m/-schema-schema`. It is used for two concrete Schema types, `:schema` and `::m/schema`. The latter is used to mark eager schema references. Quick fix would be to allow them both in line 238, but this is not future-proof as users can define their own reference schema-types which can't contribute to the internal `if` we have here.

I'm ok with the quick fix here (with test), but if there is a simple way to make this future-proof, I'm all ears.

Reply:
Arguably a user might be expecting the behaviour from your "failing case". Perhaps they want to name regex fragments:

```clj
[:cat {:registry {::option-and-value [:cat keyword? [:maybe int?]]}}
 ::option-and-value
 ::option-and-value]
```
vs.
```clj
[:cat {:registry {::option-and-value [:cat keyword? [:maybe int?]]}}
 [:schema ::option-and-value]
 [:schema ::option-and-value]]
```

Reply:
so, options:

1. add the `::m/schema` to the if and this fix is good enough for now
2. better way to escape the sequences

for 2, what do you think of the following:
* continue to use `:schema` as (the default) sequence schema escaper
* mark `::m/schema` as a regex-op (BREAKING)

with this, the code would simplify a lot:
* no need for `-re-cat-min-max` => the responsibility to handle escaping the sequence is purely on the Schema implementation
* would support the fragments you presented

I think the only change needed would be to change the following `-schema-schema` `RegexSchema` methods:

```clojure
(-regex-op? [_] internal)

(-regex-min-max [_]
  (if internal
    (-regex-min-max child)
    {:min 1, :max 1}))
```

Would be a breaking change, but smaller than dropping the `:schema` as an way to escape the sequence and for the better (allowing "fragments").

WDYT?